File: C++/Easy/Accepted/divide-a-string-into-groups-of-size-k_5.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        
        int n = s.size();
        int m = (n + k - 1) / k;
        vector<string> res(m);
        int i = 0;
        for (int j = 0; j < m; ++j) {
            res[j] = s.substr(i, k);
            if (res[j].size() < k) {
                res[j].append(k - res[j].size(), fill);
            }
            i += k;
        }
        return res;
    }
};

File: Rust/Easy/Accepted/divide-a-string-into-groups-of-size-k_5.rs
impl Solution {
    pub fn divide_string(s: String, k: i32, fill: char) -> Vec<String> {
        
        let mut result = vec![];
        let mut s = s.chars().collect::<Vec<char>>();
        let mut i = 0;
        while i < s.len() {
            let mut group = vec![];
            for _ in 0..k {
                if i < s.len() {
                    group.push(s[i]);
                } else {
                    group.push(fill);
                }
                i += 1;
            }
            result.push(group.iter().collect::<String>());
        }
        result
    }
}